sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -110,7 +112,7 @@ class QueryExecution(
     executedPlan.execute(), sparkSession.sessionState.conf)
 
   /** Get the metrics observed during the execution of the query plan. */
-  def observedMetrics: Map[String, Row] = CollectMetricsExec.collect(executedPlan)
+  def observedMetrics: java.util.Map[String, Row] = CollectMetricsExec.collect(executedPlan).asJava
```

Comment:
FYI @hvanhovell as I believe you're already using this API; this changes the method signature.